modules/util/portal-tools-sample-sql-builder/src/main/java/com/liferay/portal/tools/sample/sql/builder/DataFactory.java

```diff
@@ -5672,6 +6212,33 @@ public List<ResourcePermissionModel> newResourcePermissionModels(
 			String.valueOf(mbMessageModel.getMessageId()), _sampleUserId);
 	}
 
+	public List<ResourcePermissionModel> newResourcePermissionModels(
+		ObjectDefinitionModel objectDefinitionModel) {
+
+		return newResourcePermissionModels(
+			ObjectDefinition.class.getName(),
+			String.valueOf(objectDefinitionModel.getObjectDefinitionId()),
+			_sampleUserId);
+	}
+
+	public List<ResourcePermissionModel> newResourcePermissionModels(
+		ObjectEntryModel objectEntryModel) {
+
+		return newResourcePermissionModels(
+			ObjectDefinition.class.getName() + "#" +
```

Comment:
Can you change this code to use objectEntryModel.getModelClassName(), we centralize this logic in this method. More context on this https://github.com/liferay-bpm/liferay-portal/pull/3782.

Reply:
We can't use this method, because we do not have access to local service on DataFactory. So it would throw a `NullPointerException`. But I changed it to use the new prefix constant.